src/compare-page-pull-request/compare-page-pull-request.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+import { h } from 'dom-chef';
+import elementReady from 'element-ready';
+import { getRepoURL } from '../page-detect';
+import 'selector-observer';
+
+const Constants = {
+    COMPARE_DETAILS_SELECTOR: '#branch-compare.content-container'
+};
+
+export default async function init() {
+    await addUrlIfComparing();
+
+    /*
+     * Changes to the compare details selector indicate that:
+     *  1) The Compare button has been hit for the first time
+     *  2) The branches that are being compared have changed
+     */
+    document.body.observeSelector(
+        Constants.COMPARE_DETAILS_SELECTOR,
+        addUrlIfComparing
+    );
+}
+
+async function addUrlIfComparing() {
+    await elementReady(Constants.COMPARE_DETAILS_SELECTOR);
+    if (isComparing()) {
+        const [source, destination] = isComparing();
+        addPRLink(source, destination);
+    }
+}
+
+export function addPRLink(source, destination, comparePage = document.body) {
+    // Get compare branches area
+    const detailSummarySection = comparePage.querySelector(
+        'ul.detail-summary--section'
+    );
+    const url = getPullRequestUrl(source, destination);
+
+    // Remove Create PR link URL if it already exists
+    const previousLink = detailSummarySection.querySelector(
+        '.js-pr-create-item'
+    );
+    if (previousLink) {
+        previousLink.parentNode.removeChild(previousLink);
+    }
+
+    const link = (
+        <li className="detail-summary--item js-pr-create-item">
+            <span
+                className="aui-icon aui-icon-small aui-iconfont-devtools-pull-request detail-summary--icon"
+                title="Pull requests"
+            >
+                Pull requests
+            </span>
+            <a href={url} title="Create a pull request">
+                Create pull request
+            </a>
+        </li>
+    );
+    detailSummarySection.appendChild(link, detailSummarySection);
+    return link;
+}
+
+function getPullRequestUrl(source, destination) {
+    return `https://bitbucket.org/${getRepoURL()}/pull-requests/new?source=${encodeURIComponent(
+        source
+    )}&dest=${encodeURIComponent(destination)}`;
+}
+
+function isComparing() {
+    // This element contains all metadata about any current compare
+    const branchCompare = document.querySelector(
+        Constants.COMPARE_DETAILS_SELECTOR
+    );
+
+    // Get source and destination branch from data attributes
+    if (
+        branchCompare &&
+        branchCompare.getAttribute('data-compare-source-value') &&
+        branchCompare.getAttribute('data-compare-dest-value')
+    ) {
+        return [
+            branchCompare.getAttribute('data-compare-source-value'),
+            branchCompare.getAttribute('data-compare-dest-value')
+        ];
+    }
+
+    return false;
```

Comment:
Can this ever actually return false? In what cases?

Reply:
Nevermind, I get your intention. What prompted me to ask was the fact that the name of the function implies that you are going to return just a boolean, when in reality you are also returning data from it. I think it would be better if we just use it to get the data, and delegate the condition to the calling function, something like this:

```js
function addUrlIfComparing() {
	const { source, destination } = getSourceAndDestination();
	if (source && destination) {
        addPRLink(source, destination);
	}
}

// ...

function getSourceAndDestination() {
    // This element contains all metadata about any current compare
    const branchCompare = document.querySelector(
        COMPARE_DETAILS_SELECTOR
    ) || { dataset: {} };

    // Get source and destination branch from data attributes
    const { compareSourceValue, compareDestValue } = branchCompare.dataset;

    return { source: compareSourceValue, destination: compareDestValue };
}
```

Something else that's nicer with this approach is that we don't have to call it twice (one to test the condition, and one to capture the data) like in the previous implementation.

P.S.: Also notice how I used the neat `.dataset` property of the DOM element ([see here](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset)).

Reply:
Much nicer thanks, `.dataset` is nice!